addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -221,6 +233,8 @@ def write(self, values):
                 raise UserError(_('You cannot set an archived employee to the existing timesheets.'))
         if 'name' in values and not values.get('name'):
             values['name'] = '/'
+        if 'company_id' in values and not values.get('company_id'):
+            del values['company_id']
```

Comment:
The problem is that in some cases when a write is called, the timesheet_preprocess method fills the vals[company_id] with False (e.a when a task is switched to a project with no companny_id). So i'm adding this security to ensure that the company_id of the timesheet is never updated to False (which would just trigger a db error anyway, since the field is required).
I've put it in the write method and not in the preprocess because it is a behavior that can only be encountered when the user does actions that triggers a write on a existing ts, not on a create.